transport/src/main/java/io/netty/channel/local/LocalChannel.java

```diff
@@ -37,12 +38,17 @@
 import java.nio.channels.ConnectionPendingException;
 import java.nio.channels.NotYetConnectedException;
 import java.util.Queue;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
 
 /**
  * A {@link Channel} for the local transport.
  */
 public class LocalChannel extends AbstractChannel {
 
+    @SuppressWarnings({ "rawtypes" })
+    private static final AtomicReferenceFieldUpdater<LocalChannel, Future> finishReadFutureUpdater =
+            AtomicReferenceFieldUpdater.newUpdater(LocalChannel.class, Future.class, "finishReadFuture");
```

Comment:
Please try to use `PlatformDependent.newAtomicReferenceFieldUpdater` first in a static block like we do in other places like:

https://github.com/netty/netty/blob/master/transport/src/main/java/io/netty/channel/ChannelOutboundBuffer.java#L90